src/plugins/upf/upf_pfcp_api.c

```diff
@@ -855,6 +856,21 @@ static int handle_create_pdr(upf_session_t *sess, pfcp_create_pdr_t *create_pdr,
 	      break;
 	    }
 	}
+
+	if (ISSET_BIT(pdr->pdi.grp.fields, PDI_APPLICATION_ID))
+		{
+			create->pdi.fields |= F_PDI_APPLICATION_ID;
+
+			pfcp_application_id_t *app_id;
+	
+			vec_foreach(app_id, pdr->pdi.application_id)
+				{
+					vec_add1(create->pdi.app_id, *app_id);
```

Comment:
vec_add1 almost immediately followed by a vec_free feels wrong. Something must be wrong here....

Reply:
My idea is to use a temporary vector as a parameter. The vector contains a list of app ids.